apps/emqx_authn/src/simple_authn/emqx_authn_http.erl

```diff
@@ -262,21 +264,39 @@ transform_header_name(Headers) ->
     ).
 
 check_ssl_opts(Conf) ->
-    {BaseUrl, _Path, _Query} = parse_url(get_conf_val("url", Conf)),
-    case BaseUrl of
-        <<"https://", _/binary>> ->
-            case get_conf_val("ssl.enable", Conf) of
-                true -> ok;
-                false -> false
-            end;
-        <<"http://", _/binary>> ->
-            ok
+    case get_conf_val("url", Conf) of
+        undefined ->
+            ok;
+        Url ->
+            {BaseUrl, _Path, _Query} = parse_url(Url),
+            case BaseUrl of
+                <<"https://", _/binary>> ->
+                    case get_conf_val("ssl.enable", Conf) of
+                        true ->
+                            ok;
+                        false ->
+                            <<"it's required to enable the TLS option to establish a https connection">>
+                    end;
+                <<"http://", _/binary>> ->
+                    ok
```

Comment:
Mb add default case? Like
```
_ -> <<"unsupported schema">>
```

Reply:
`parse_url(Url)` will throw error when mean unspported schema. 
so we don't  need a default case.